db-tests/src/test/java/gov/nasa/jpl/aerie/database/ActivityDirectiveChangelogTests.java

```diff
@@ -0,0 +1,310 @@
+    res.close();
+
+    assertEquals(expectedCount, revision);
+  }
+
+  @Test
+  void shouldCreateChangelogForUpdatedActDir() throws SQLException {
+    final var expectedOffsetCurrent = "01:00:00";
+    final var expectedOffsetPrevious = "00:00:00";
+    final var expectedCountBefore = 1;
+
+    final var planId = merlinHelper.insertPlan(
+        missionModelId,
+        merlinHelper.admin.name(),
+        "changelog plan" + UUID.randomUUID(),
+        "1970-1-1 00:00:00+00");
+    final var activityId = merlinHelper.insertActivity(planId);
+
+    final var resBefore = connection.createStatement()
+                                    .executeQuery(
+                                        // language=sql
+                                        """
+                                            SELECT count(revision)
+                                            FROM activity_directive_changelog
+                                            WHERE plan_id = %d and activity_directive_id = %d;"""
+                                            .formatted(planId, activityId)
+                                    );
+
+    resBefore.next();
+    assertEquals(expectedCountBefore, resBefore.getInt(1));
+    resBefore.close();
+
+    final var updatedRows = connection.createStatement()
+                                      .executeQuery(
+                                          // language=sql
+                                          """
+                                              UPDATE activity_directive
+                                              SET start_offset = '%s'
+                                              WHERE plan_id = %d and id = %d
+                                              RETURNING id;"""
+                                              .formatted(expectedOffsetCurrent, planId, activityId)
+                                      );
+    updatedRows.next();
+    assertEquals(activityId, updatedRows.getInt(1));
+    updatedRows.close();
+
+    final var resAfter = connection.createStatement()
+                                   .executeQuery(
+                                       // language=sql
+                                       """
+                                           SELECT revision, to_char(start_offset, 'HH24:MI:SS') start_offset
+                                           FROM activity_directive_changelog
+                                           WHERE plan_id = %d and activity_directive_id = %d
+                                           ORDER BY revision DESC;"""
+                                           .formatted(planId, activityId)
+                                   );
+
+    resAfter.next();
+    final var startNew = resAfter.getString("start_offset");
+    final var revNew = resAfter.getInt("revision");
+    resAfter.next();
+    final var startOld = resAfter.getString("start_offset");
+    final var revOld = resAfter.getInt("revision");
+    resAfter.close();
+
+    assertTrue(revOld < revNew);
```

Comment:
We know for sure that `revOld == 1` and `revNew == 2`, right? If so, assert that more specific fact instead of `revOld < revNew`. Otherwise revOld = -1 and revNew = 1000 would pass this check, even though we know that's not what we expect them to be.

Reply:
Good point, that check was left behind from when `revision` was unique within the whole DB, so only the ordering (not specific values) was known.